File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # By using a hashmap (dictionary in case of python)
        hash_map = {}

        for index1, value in enumerate(nums):
            req_value = target - value

            if req_value in hash_map:
                index2 = hash_map[req_value]
                return [index1, index2]
            
            hash_map[value] = index1

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numsMap = new HashMap<>();
        int n = nums.length;

        for(int i = 0; i < n; i++) {
            int diffNeed = target - nums[i];
            
            if (numsMap.containsKey(diffNeed)) {
                return new int[]{numsMap.get(diffNeed), i};
            }

            numsMap.put(nums[i], i);
        }

        return new int[]{};
    }
}